airflow/providers/snowflake/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,153 @@
+    :param warehouse: name of warehouse (will overwrite any warehouse
+        defined in the connection's extra JSON)
+    :type warehouse: str
+    :param database: reference to a specific database in Snowflake connection
+    :type database: str
+    :param snowflake_conn_id: Reference to
+        :ref:`Snowflake connection id<howto/connection:snowflake>`
+    :type snowflake_conn_id: str
+    :param role: name of role (will overwrite any role defined in
+        connection's extra JSON)
+    :type role: str
+    :param authenticator: authenticator for Snowflake.
+        'snowflake' (default) to use the internal Snowflake authenticator
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :type authenticator: str
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :type session_parameters: dict
+    """
+
+    def __init__(
+        self,
+        *,
+        stage: str,
+        prefix: Optional[str] = None,
+        file_format: str,
+        on_error: Optional[str] = None,
+        unload_sql: str,
+        header: Optional[bool] = False,
+        single: Optional[bool] = False,
+        overwrite: Optional[bool] = False,
+        schema: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        database: Optional[str] = None,
+        autocommit: bool = True,
+        snowflake_conn_id: str = 'snowflake_default',
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.warehouse = warehouse
+        self.database = database
+        self.stage = stage
+        self.prefix = prefix
+        self.file_format = file_format
+        self.on_error = on_error
+        self.unload_sql = unload_sql
+        self.header = header
+        self.single = single
+        self.overwrite = overwrite
+        self.schema = schema
+        self.autocommit = autocommit
+        self.snowflake_conn_id = snowflake_conn_id
+        self.role = role
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
+        self.query_ids: List[str] = []
+
+    def execute(self, context: Any) -> None:
+        snowflake_hook = SnowflakeHook(
+            snowflake_conn_id=self.snowflake_conn_id,
+            warehouse=self.warehouse,
+            database=self.database,
+            role=self.role,
+            schema=self.schema,
+            authenticator=self.authenticator,
+            session_parameters=self.session_parameters,
+        )
+
+        sql_parts = [
+            f"COPY INTO @{self.stage}/{self.prefix or ''}",
```

Comment:
Stage is the identifier of the object, so it can contain spaces. We should try to ensure that this identifier is safely passed to the query being built, e.g. the use of spaces in the identifier does not cause problems. See: https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html

Reply:
Here we expect user to pass valid identifier, anyway do you expect SnowflakeToS3Operator to validate stage name using some regex?